accord-core/src/main/java/accord/impl/InMemoryCommandStore.java

```diff
@@ -325,9 +325,18 @@ public boolean hasCommandsForKey(RoutingKey key)
     @Override
     protected void ensureDurable(Ranges ranges, RedundantBefore onCommandStoreDurable)
     {
-        ((InMemoryAgent)agent).snapshot(this).begin((success, fail) -> execute((Empty)() -> "Report CommandStore Durable", safeStore -> {
-            safeStore.upsertRedundantBefore(onCommandStoreDurable);
-        }).begin(agent));
+        // TODO (required): we need a more general Replaying state
+        if (!CommandsForKey.reportLinearizabilityViolations())
```

Comment:
hah, yes, maybe we should add a static flag and use it in CommandsForKey.reportLinearizabilityViolations?